### ISSUE TITLE
Inconsistent order of arguments in loss functions --> potential issue for dual contrastive loss

### ISSUE THREAD

**Issue by JoMe2704** [lucidrains/stylegan2-pytorch#289]
Hi,
I observed that hinge_loss and gen_hinge_loss use a different oder of inputs:
`def gen_hinge_loss(fake, real):`
`def hinge_loss(real, fake): `

As a consequence, D_loss_fn and G_loss_fn are also invoked with different order of inputs:
`divergence = D_loss_fn(real_output_loss, fake_output_loss)`
`loss = G_loss_fn(fake_output_loss, real_output)`

When changing to dual contrastive loss, this will result in an issue, because both D_loss_fn and G_loss_fn are then implemented by the same function dual_contrastive_loss, which assumes the order
`dual_contrastive_loss(real_logits, fake_logits)`
Consequently, when using dual contrastive loss, the line
`loss = G_loss_fn(fake_output_loss, real_output)`
uses the wrong order.

Do I miss something, here?

**Comment by lucidrains** [comment]
@JoMe2704 ah nice catch

i think that dual contrastive loss is the same whether the fake and real is swapped or not, so it is ok. but let me know if it isn't

**Comment by JoMe2704** [comment]
You mean it's symmetric w.r.t to its inputs? No, it's not. If it was, it wouldn't help the discriminator learn to distinguish between real and fake. 
I also tested it, it's definitely not symmetric. The order of inputs matters.

**Comment by lucidrains** [comment]
@JoMe2704 ah, my bad

should be fixed 🙏
